src/main/java/com/example/demo/controller/OrderController.java

```diff
@@ -0,0 +1,75 @@
+package com.example.demo.controller;
+
+import com.example.demo.dto.order.CreateOrderRequestDto;
+import com.example.demo.dto.order.OrderDto;
+import com.example.demo.dto.order.UpdateOrderStatusDto;
+import com.example.demo.dto.orderitem.OrderItemDto;
+import com.example.demo.model.User;
+import com.example.demo.service.order.OrderService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Order manager", description = "Endpoints for managing orders")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+
+    @Operation(summary = "Create order")
+    @PreAuthorize("hasRole('ROLE_USER')")
```

Comment:
it is not necessary to check for USER role if you have only USER and ADMIN roles